tests/unit/test_template.py

```diff
@@ -113,3 +114,31 @@ def test_check_renderer_blacklisting(self):
             "jinja|json", self.render_dict, ["jinja"], ["jinja", "json"]
         )
         self.assertListEqual([("fake_json_func", "")], ret)
+
+    def test_check_renderer_py_warning(self):
+        """
+        Check that warning is emitted if "py" has preceding renderers.
+        """
+        warn = "ignores previously processed results"
+        mock_log = MagicMock()
+        with patch.object(template.log, "warning", mock_log):
```

Comment:
Not sure, but would it be possible to dynamically grab the list of renderers from somewhere within Salt? I think we should at minimum show jinja/json (or, assert that there are at least one renderer in the list), but that would ensure that we're testing any potential future renderers.

If that's not possible with a reasonable amount of effort, then feel free to ignore the suggestion :slightly_smiling_face:

Reply:
@waynew Can you help to improve this PR instead (it supersedes mine)? https://github.com/saltstack/salt/pull/55390

Reply:
Should we go ahead and close this one then?

Reply:
I’ll close mine once the other one is merged. That is why I set the title to WIP